From 556e87f5574f4a51477ad451ac5228654f6d3d11 Mon Sep 17 00:00:00 2001
From: Supreeth Venkatesh <supreeth.venkatesh@amd.com>
Date: Sun, 23 May 2021 13:08:51 -0500
Subject: [PATCH 1/4] Add Initial u-boot configuration
Content-Type: text/plain; charset="us-ascii"
Content-Transfer-Encoding: 7bit

Adds ast2600-hawaii device tree and configs based on ast2600 OpenBMC
defconfig.

Signed-off-by: Supreeth Venkatesh <supreeth.venkatesh@amd.com>
---
 arch/arm/dts/Makefile             |   1 +
 arch/arm/dts/ast2600-hawaii.dts   | 188 ++++++++++++++++++++++++++++++
 configs/ast2600_openbmc_defconfig |   6 +-
 3 files changed, 193 insertions(+), 2 deletions(-)
 create mode 100644 arch/arm/dts/ast2600-hawaii.dts

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 786042cd83..89531ada40 100755
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -682,6 +682,7 @@ dtb-$(CONFIG_ARCH_ASPEED) += \
 	ast2600a0-evb.dtb \
 	ast2600a1-evb.dtb \
 	ast2600-fpga.dtb \
+	ast2600-hawaii.dtb \
 	ast2600-rainier.dtb \
 	ast2600-slt.dtb \
 	ast2600-tacoma.dtb
diff --git a/arch/arm/dts/ast2600-hawaii.dts b/arch/arm/dts/ast2600-hawaii.dts
new file mode 100644
index 0000000000..83cfa66dd2
--- /dev/null
+++ b/arch/arm/dts/ast2600-hawaii.dts
@@ -0,0 +1,188 @@
+/dts-v1/;
+
+#include "ast2600-u-boot.dtsi"
+
+/ {
+	model = "AST2600 HAWAII";
+	compatible = "aspeed,ast2600-hawaii", "aspeed,ast2600";
+
+	memory {
+		device_type = "memory";
+		reg = <0x80000000 0x80000000>;
+	};
+
+	chosen {
+		stdout-path = &uart5;
+		bootargs = "console=ttyS4,115200n8 root=/dev/ram rw vmalloc=512MB";
+	};
+
+	aliases {
+		mmc1 = &sdhci_slot0;
+		spi0 = &fmc;
+		spi1 = &spi1;
+		spi2 = &spi2;
+		ethernet0 = &mac3;
+	};
+
+	cpus {
+		cpu@0 {
+			clock-frequency = <800000000>;
+		};
+		cpu@1 {
+			clock-frequency = <800000000>;
+		};
+	};
+};
+
+&uart5 {
+	u-boot,dm-pre-reloc;
+	status = "okay";
+};
+
+&sdrammc {
+	clock-frequency = <400000000>;
+};
+
+&wdt1 {
+	status = "okay";
+};
+
+&wdt2 {
+	status = "okay";
+};
+
+&wdt3 {
+	status = "okay";
+};
+
+&mdio {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_mdio1_default>;
+
+	ethphy0: ethernet-phy@0 {
+		reg = <0>;
+	};
+};
+
+&mac3 {
+	status = "okay";
+	phy-mode = "rgmii";
+	phy-handle = <&ethphy0>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_rgmii4_default &pinctrl_mac4link_default>;
+};
+
+// BMC FLASH
+&fmc {
+	status = "okay";
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_fmcquad_default>;
+
+	flash@0 {
+		compatible = "spi-flash";
+		status = "okay";
+		spi-max-frequency = <50000000>;
+		spi-tx-bus-width = <4>;
+		spi-rx-bus-width = <4>;
+	};
+};
+
+// FPGA and BIOS FLASH
+
+&spi1 {
+	status = "okay";
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_spi1_default &pinctrl_spi1quad_default>;
+
+	flash@0 {
+		compatible = "spi-flash";
+		status = "okay";
+		spi-max-frequency = <50000000>;
+		spi-tx-bus-width = <4>;
+		spi-rx-bus-width = <4>;
+	};
+};
+
+&sdhci {
+	timing-phase = <0xc6ffff>;
+};
+
+// TODO: Re-check pwr-gpios and pwr-sw-gpios
+&sdhci_slot0 {
+	status = "okay";
+	bus-width = <4>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_sd1_default>;
+	sdhci-drive-type = <1>;
+};
+
+&i2c4 {
+	status = "okay";
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c5_default>;
+};
+
+&i2c5 {
+	status = "okay";
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c6_default>;
+};
+
+&i2c6 {
+	status = "okay";
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c7_default>;
+};
+
+&i2c7 {
+	status = "okay";
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c8_default>;
+};
+
+&i2c8 {
+	status = "okay";
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c9_default>;
+};
+
+&i2c14 {
+	status = "okay";
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c15_default>;
+};
+
+&pcie_bridge1 {
+	status = "okay";
+};
+
+&h2x {
+	status = "okay";
+};
+
+&ehci1 {
+	status = "okay";
+};
+
+&display_port {
+	status = "okay";
+};
+
+&scu {
+	mac3-clk-delay = <0x0a 0x04
+			  0x08 0x04
+			  0x08 0x04>;
+};
+
+&hace {
+	status = "okay";
+};
diff --git a/configs/ast2600_openbmc_defconfig b/configs/ast2600_openbmc_defconfig
index 2e2df2e3a2..8b5bd5730d 100755
--- a/configs/ast2600_openbmc_defconfig
+++ b/configs/ast2600_openbmc_defconfig
@@ -11,7 +11,7 @@ CONFIG_ARMV7_BOOT_SEC_DEFAULT=y
 CONFIG_NR_DRAM_BANKS=1
 CONFIG_FIT=y
 CONFIG_USE_BOOTARGS=y
-CONFIG_BOOTARGS="console=ttyS4,115200n8 root=/dev/ram rw"
+CONFIG_BOOTARGS="console=ttyS4,115200n8 root=/dev/ram rw vmalloc=512MB"
 CONFIG_USE_BOOTCOMMAND=y
 CONFIG_BOOTCOMMAND="bootm 20100000"
 CONFIG_SYS_CONSOLE_ENV_OVERWRITE=y
@@ -19,7 +19,7 @@ CONFIG_DISPLAY_BOARDINFO_LATE=y
 CONFIG_ARCH_EARLY_INIT_R=y
 CONFIG_HUSH_PARSER=y
 # CONFIG_AUTO_COMPLETE is not set
-CONFIG_SYS_PROMPT="ast# "
+CONFIG_SYS_PROMPT="hawaii# "
 CONFIG_CMD_BOOTZ=y
 # CONFIG_CMD_BOOTEFI is not set
 # CONFIG_CMD_ELF is not set
@@ -44,6 +44,7 @@ CONFIG_CMD_EXT4=y
 CONFIG_CMD_EXT4_WRITE=y
 CONFIG_CMD_FAT=y
 CONFIG_CMD_FS_GENERIC=y
+CONFIG_MTD_PARTITIONS=y
 CONFIG_CMD_MTDPARTS=y
 CONFIG_EFI_PARTITION=y
 CONFIG_ENV_IS_IN_SPI_FLASH=y
@@ -74,6 +75,7 @@ CONFIG_SPI_FLASH_MACRONIX=y
 CONFIG_SPI_FLASH_SPANSION=y
 CONFIG_SPI_FLASH_STMICRO=y
 CONFIG_SPI_FLASH_WINBOND=y
+CONFIG_SPI_FLASH_MTD=y
 CONFIG_PHY_BROADCOM=y
 CONFIG_PHY_REALTEK=y
 CONFIG_PHY_NCSI=y
--
2.17.1

